types.go

```diff
@@ -664,6 +669,13 @@ type StringOrInt int
 
 func (d *StringOrInt) UnmarshalJSON(b []byte) error {
 	str := strings.Replace(string(b), "\"", "", -1)
+
+	numeric := regexp.MustCompile(digitOnlyRegex).MatchString(str)
```

Comment:
ya so regexp.MustCompile need to be a package var as it's extra work to do a MustCompile ever call to Unmarshall. 
Do something like this instead of the const above...

```
var numericRegex = regexp.MustCompile(`\d`)
```

and in the Unmarshall call `numericRegex.MatchString(str)` this way it's compiled once for both unmarshalls at runtime... and technically MustCompile can panic https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/regexp/regexp.go;l=319 so it's better practice to use it at runtime instead of call time.

Reply:
This is a lot more logical than what I original undestood :)